sympy/tensor/array/expressions/array_expressions.py

```diff
@@ -1458,6 +1459,35 @@ def as_explicit(self):
         return tensorcontraction(self.expr.as_explicit(), *self.contraction_indices)
 
 
+class Reshape(_CodegenArrayAbstract):
+
+    def __new__(cls, expr, shape):
+        expr = _sympify(expr)
+        if not isinstance(shape, Tuple):
+            shape = Tuple(*shape)
```

Comment:
Do I understand correctly that the object is created even if the new shape does not match the original shape?

```python
Eq(Mul(*expr.shape), Mul(*shape)) != False
```
Should allow any unknown symbolic sizes, but would not allow e.g. 
```python
from sympy import Eq, Symbol, Mul
x= Symbol('x', positive = True)
shape1=(x,)
shape2=(x,2)
Eq(Mul(*shape1),Mul(*shape2)).doit() != False
```

Reply:
> Do I understand correctly that the object is created even if the new shape does not match the original shape?

Yes, I will add a check.